packages/solana/lib/src/rpc_client/account.dart

```diff
@@ -69,11 +82,61 @@ class AccountData with _$AccountData {
         throw FormatException('unexpected data "${data.first}"');
       }
       return AccountData.fromBytes(base64Decode(base64String));
-    } else if (data is Map) {
-      // TODO(IA): object data, must be converted into a valid thing
-      return const AccountData.empty();
+    } else if (data is Map<String, dynamic>) {
+      switch (data['program']) {
+        // TODO(IA): add other program data
+        case 'spl-token':
+          return AccountData.splToken(ParsedSplTokenAccountData.fromJson(
+            data['parsed'] as Map<String, dynamic>,
+          ));
+        default:
+          return AccountData.generic(data);
+      }
     } else {
       throw const FormatException('cannot decode account data');
     }
   }
+
+  @override
+  Map<String, dynamic> toJson(AccountData? object) => <String, dynamic>{};
```

Comment:
Since it's not actually intended for usage, it should rather throw `UnsupportedError`.